pkg/cmd/server/start.go

```diff
@@ -70,16 +70,36 @@ type PostStartHook struct {
 	Name string
 }
 
+type StartOptions struct {
+	EtcdPath         string
+	Apis             []*builders.APIGroupBuilder
+	Openapidefs      openapi.GetOpenAPIDefinitions
+	Title            string
+	Version          string
+	TweakConfigFuncs []func(apiServer *apiserver.Config) error
+
+	//FlagConfigFunc handles user-defined flags
+	FlagConfigFuncs []func(*cobra.Command) error
+}
+
 // StartApiServer starts an apiserver hosting the provider apis and openapi definitions.
-func StartApiServer(etcdPath string, apis []*builders.APIGroupBuilder, openapidefs openapi.GetOpenAPIDefinitions, title, version string, tweakConfigFuncs ...func(apiServer *apiserver.Config) error) {
+func StartApiServer(opts *StartOptions) {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	GetOpenApiDefinition = openapidefs
+	GetOpenApiDefinition = opts.Openapidefs
 
 	signalCh := genericapiserver.SetupSignalHandler()
 	// To disable providers, manually specify the list provided by getKnownProviders()
-	cmd, _ := NewCommandStartServer(etcdPath, os.Stdout, os.Stderr, apis, signalCh, title, version, tweakConfigFuncs...)
+	cmd, _ := NewCommandStartServer(opts.EtcdPath, os.Stdout, os.Stderr, opts.Apis, signalCh,
+		opts.Title, opts.Version, opts.TweakConfigFuncs...)
+
+	for _, ff := range opts.FlagConfigFuncs {
+		if err := ff(cmd); err != nil {
+			panic(err)
```

Comment:
how about propagating/returning the error to make the panic opt-out?

Reply:
Do you mean we should return error?

Reply:
yea, error